services/src/main/scala/cromwell/pipeline/service/ProjectService.scala

```diff
@@ -40,22 +44,34 @@ object ProjectService {
     new ProjectService {
 
       def getUserProjectById(projectId: ProjectId, userId: UserId): Future[Project] =
-        projectRepository.getProjectById(projectId).flatMap(project => getForUserOrFail(project.toSeq, userId))
+        projectRepository
+          .getProjectById(projectId)
+          .flatMap(project => getForUserOrFail(project.toSeq, userId))
+          .recoverWith {
+            case exc: ServiceException => Future.failed(exc)
+            case _: Exception =>
+              Future.failed(InternalError("Failed to find user project due to unexpected internal error"))
+          }
```

Comment:
```suggestion
        projectRepository
          .getProjectById(projectId)
          .recoverWith {
            case _ =>  Future.failed(InternalError("Failed to find user project due to unexpected internal error"))
          }.flatMap(project => getForUserOrFail(project.toSeq, userId))
```
Make your recoveries as close as possible to places that can return an error. You won't have to take into account existing service exception and it's much easier to understand what error you are trying to recover from. Same for other places (except for `addProject` and `updateProjectName`, they look fine)

Reply:
Thanks for clarification! I'll change it.

Reply:
Done.